feat: Update to Hono v3.0.0

lrazovic opened the pull request:
This PR updates the GraphQL middleware to [Hono v3.0.0](https://github.com/honojs/hono/releases/tag/v3.0.0).

Overview of the changes:
- Update Hono dependencies from v2.7.5 to v.3.0.0
- ~`Requests` is now `HonoRequest` [src](https://github.com/honojs/hono/blob/main/docs/MIGRATION.md#creq-is-now-honorequest)~
- ~Update `graphql@16.6.0` import from `cdn.skypack.dev` to `esm.sh` due to Deno incompatibility~

EDIT:

- For this PR I think a version bump is needed

yusukebe commented:
Hi @lrazovic 

Thank you for your PR.

I have a few requests.

* Update the npm package, including `src` first. We also may fix `peerDependencies`.
* Then run the `yarn denoify` to generate files for Deno. The files in `deno_dist` should not be written manually.

lrazovic commented:
Thanks for the feedback @yusukebe, I learned a lot about [denoify](https://github.com/garronej/denoify)!

- I updated to `"hono": "^3.0.0"` the `peerDependencies` and `devDependencies`.
- Bumped the package version from `0.2.0` to `0.3.0`
- Updated to `https://deno.land/x/hono@v3.0.0/mod.ts` in `deno_test`
- Run `yarn prerelease` to build,  denoify and test:all

yusukebe commented:
Thanks, @lrazovic !

> Let me know if it's ok to get the Request using c.req.raw or you prefer to pass the HonoRequest to getGraphQLParams, parseBody and parseFormURL.

I think it's OK to use `c.req.raw`!

yusukebe commented:
@lrazovic 

The last step. Could you do `yarn changeset` and write the change?

yusukebe commented:
I'll merge it!
